src/start/qemu.md

```diff
@@ -433,6 +440,26 @@ also running the embedded program.
 
 In this section we'll use the `hello` example we already compiled.
 
+Before getting started remotely debugging our app running in QEMU with GDB,
+let's modify our app configuration `.cargo/config`, to enable the GNU ARM linker: 
+
+```toml
+...
+rustflags = [
+  # LLD (shipped with the Rust toolchain) is used as the default linker
+  #"-C", "link-arg=-Tlink.x",  <--- Comment this line
```

Comment:
It's still required to have `link-arg=-Tlink.x`, otherwise the linker script isn't used at all. The user would have to just add `linker=arm-none-eabi-ld` while keeping `link-arg=-Tlink.x`.

That said, I'm not sure why changing linker helps in this case anyway, which might be worth a little more investigation.

Reply:
I've seen a handful of bugs where LLD messes up debuginfo sections, but I don't think we should recommend GNU LD by default because of that. LLD is the default Rust linker for thumb targets.

Reply:
I don't know how much the debug info is messed up. The application compiled with LLD works, stepping through the code with gdb works, at least for that minimalist hello world.
So perhaps it's enough for the time being to just signal in the book the bug with smtg like:

> once connected to the remote qemu, gdb should indicate that it is paused at the reset() function, waiting for your input.
> `Reset () at $REGISTRY/cortex-m-rt-0.6.1/src/lib.rs:473`
> 
>  However in some setup, gdb may instead emit some warnings like : `core::num::bignum::Big32x40::mul_small () at src/libcore/num/bignum.rs:254
> 254     src/libcore/num/bignum.rs: No such file or directory.
> `
> You can safely ignore those warnings. 

I can file a bug for that linker issue, I would appreciate your help if you can suggest where the bug should be filed.

Reply:
I updated the PR, removing the paragraph recommanding to switch to the GNU LD. 

I added some text hinting at the possible debuginfo mix-up glitch, and that it can be safely ignored.